src/reducers/tasksReducer.js

```diff
@@ -163,6 +164,42 @@ export default function(state = initialState, action) {
       });
     }
 
+    case 'MOVE_TASK':
+      const { oldParentGid, newParentGid, taskGid } = action;
+      const newMap = Object.assign({}, state.taskMap);
+      const newParentTask = newMap[newParentGid];
+      const oldParentTask = newMap[oldParentGid];
+      const oldParentSubtaskGids = oldParentTask.subtask_gids.filter(
+        gid => gid !== taskGid
+      );
+      const newSubtaskGids = [...newParentTask.subtask_gids, taskGid];
+      const newSubtasks = newSubtaskGids.filter(
+        (gid, idx) => newSubtaskGids.indexOf(gid) === idx
+      );
+
+      newSubtasks.sort(function(gid1, gid2) {
+        const t1Date = new Date(newMap[gid1].created);
+        const t2Date = new Date(newMap[gid2].created);
```

Comment:
@colinfruit Efficiency idea:

```
const t1 = newMap[gid1];
const t2 = newMap[gid2];
t1.created_parsed = t1.created_parsed || new Date(t1.created);
t2.created_parsed = t2.created_parsed || new Date(t2.created);
```

so that there's no need to parse any task's `created` timestamp twice.

Reply:
nice, created_parsed a field? Or is it on the todo list?

Reply:
It's a field that we'd be creating just to make this more efficient.

Reply:
The point is that, the second time that one of these tasks's `created` timestamp goes to be parsed, it'll reuse `created_parsed` rather than parsing the `created` field again and creating a `Date` object out of it.